installer/tue-install-impl.bash

```diff
@@ -375,47 +375,74 @@ function tue-install-add-text
         tue-install-error "Invalid tue-install-add-text call. Usage: tue-install-add-text SOURCE_FILE TARGET_FILE"
     fi
 
-    local source_file=$TUE_INSTALL_CURRENT_TARGET_DIR/$1
+    tue-install-debug "tue-install-add-text $*"
+
+    local source_file=$1
+    if [[ "$source_file" != "/"* ]] || [[ "$source_file" != "~/"* ]]
+    then
+        tue-install-debug "tue-install-add-text: relative source file"
+        source_file="$TUE_INSTALL_CURRENT_TARGET_DIR/$source_file"
+    else
+        tue-install-debug "tue-install-add-text: absolute source file"
+    fi
     local target_file=$2
+    if [[ "$target_file" != "/"* ]] && [[ "$source_file" != "~/"* ]]
+    then
+        tue-install-error "tue-install-add-text: target file needs to be absolute or relative to the home directory"
+    fi
 
     local root_required=true
     if namei -l $target_file | grep -q $(whoami)
     then
+        tue-install-debug "tue-install-add-text: NO root required"
         root_required=false
+    else
+        tue-install-debug "tue-install-add-text: root required"
     fi
 
     if [ ! -f $source_file ]
     then
-        tue-install-error "tue-install-add-text: No such file: $source_file"
+        tue-install-error "tue-install-add-text: No such source file: $source_file"
     fi
 
     if [ ! -f $target_file ]
     then
-        tue-install-error "tue-install-add-text: No such file: $target_file"
+        tue-install-error "tue-install-add-text: No such target file: $target_file"
     fi
 
     local begin_tag=$(head -n 1 $source_file)
     local end_tag=$(awk '/./{line=$0} END{print line}' $source_file)
     local text=$(sed -e :a -e '/^\n*$/{$d;N;};/\n&/ba' $source_file)
+    tue-install-debug "tue-install-add-text: Lines to be added: \n$text"
```

Comment:
If root is required to read source, then `sudo` needs to be added in the above commands